LibraBFT/Concrete/Obligations/PreferredRound.agda

```diff
@@ -3,10 +3,11 @@
    Copyright (c) 2020, 2021, Oracle and/or its affiliates.
    Licensed under the Universal Permissive License v 1.0 as shown at https://opensource.oracle.com/licenses/upl
 -}
-open import LibraBFT.Prelude
-open import LibraBFT.Lemmas
 open import LibraBFT.Base.Types
-open import LibraBFT.Impl.Base.Types
+open import LibraBFT.ImplShared.Base.Types
+open import LibraBFT.Lemmas
+
+open import LibraBFT.Prelude
```

Comment:
For better or worse, our [style guide])(https://github.com/oracle/bft-consensus-agda/blob/main/STYLE.md#coding-style-conventions), it says to list common imports (like `LibraBFT.Prelude` and `Optics.All`) first, and I think we've stuck to that fairly consistently.  I know your preference (in Haskell) to list imports in alphabetical order, and I share that preference and try to follow it "to the extent possible" (also mentioned in the style guide).  But note that we cannot stick strictly to this rule (some imports depend on previous ones).  You can argue that the style guide is suboptimal, and maybe we'll agree to change it, in which case we should change everywhere to be consistent with it.  I'm in favour of keeping the style guide as-is, and reverting the changes that mix the really common imports in among others, but not "hard over" on this issue, as long as we understand that this is not part of moving towards strictly alpha-order, which we can't.

Reply:
Alpha order is better, because:
- Agda does not complain about duplicate imports (unless there is a conflict.
- So alpha makes it easier to see the duplication.

That said, some import depend on earlier imports.  In that case, they should be put below the alpha section with a blank line between.

Reply:
OK, I'm fine if you want to change the style guide along these lines and conform with  it.  I would just use slightly loser language that doesn't require strictly alphabetical followed by non alphabetical, because I sometimes want to list the import and then the one that depends on it right after.  How about alphabetical order, except where there is a dependency that requires a different order, in which case there should be a blank line between any two imports that aren't in alpha order?  Also, I really think we should go with a lighter blue for the bike shed.

Reply:
Actually, even less specific is better.  Maybe something like "use blank lines to avoid any confusion due to non-alpha order imports".

Reply:
OK.   When I update the request, please check the style guide update.